src/Course/Optional.hs

```diff
@@ -92,7 +97,10 @@ mapOptional = error "todo: Course.Optional#mapOptional"
 --     a) Is there a pattern that all of their type signatures fall into?
 --     b) Are there similarities in their implementations? Explain.
 question_Optional_2 :: (String, String)
-question_Optional_2 = error "todo"
+question_Optional_2 =
+  ( "Yes, the signature for each map is roughly the same, only differing in the name of the type in the map signature."
+  , "There are also equivalencies in the implementations, up to cardinality of data constructors. By that, I mean that the implementation for type `Foo` is usually `mapFoo f (SubFoo x) = SubFoo (f x)` for each data constructor `SubFoo` in `Foo`. There will be further differences given the needs/arguments for each data constructor (like how `Error s` disregards the function)."
+  )
```

Comment:
exactly, all `fmap` implementations are derived from the structure of the datatype in exactly the same way :-p

good.